linkerd/http/prom/src/count_reqs.rs

```diff
@@ -6,18 +8,29 @@ use prometheus_client::{
 };
 use std::task::{Context, Poll};
 
+/// A [`Family`] of counters with `L`-encoded labels.
+///
+/// See [`EncodeLabelSet`] for more information about encoding labels.
 #[derive(Clone, Debug)]
 pub struct RequestCountFamilies<L>(Family<L, Counter>);
 
+// A single [`Counter`] that tracks the number of requests.
 #[derive(Clone, Debug)]
 pub struct RequestCount(Counter);
 
+/// A [`NewService<T>`][svc::NewService] that creates [`RequestCount`] services.
 #[derive(Clone, Debug)]
 pub struct NewCountRequests<X, N> {
+    /// The inner [`NewService<T>`][svc::NewService].
     inner: N,
+    /// The [`ExtractParam<P, T>`][svc::ExtractParam] strategy for obtaining our parameters.
     extract: X,
 }
 
+/// Counts requests for an inner `S`-typed [`Service`][svc::Service].
+///
+/// This will increment its counter when [`call()`][svc::Service::call]ed, before calling the inner
```

Comment:
meta: do we lint for broken doc links?

https://doc.rust-lang.org/rustdoc/lints.html#broken_intra_doc_links

we could disallow broken links via `broken_intra_doc_links`, though i know it can sometimes be a chore to fix that if it hasn't been enforced previously.

Reply:
AFAIK we don't lint docs in application code (we probably do when we publish to crates.io).